Modules/_datetimemodule.c

```diff
@@ -6537,136 +6514,127 @@ PyInit__datetime(void)
     };
 
     for (size_t i = 0; i < Py_ARRAY_LENGTH(types); i++) {
-        if (PyModule_AddType(m, types[i]) < 0) {
-            return NULL;
+        if (PyModule_AddType(module, types[i]) < 0) {
+            return -1;
         }
     }
 
     if (PyType_Ready(&PyDateTime_IsoCalendarDateType) < 0) {
-        return NULL;
+        return -1;
     }
     Py_INCREF(&PyDateTime_IsoCalendarDateType);
 
-    /* timedelta values */
-    d = PyDateTime_DeltaType.tp_dict;
-
-    x = new_delta(0, 0, 1, 0);
-    if (x == NULL || PyDict_SetItemString(d, "resolution", x) < 0)
-        return NULL;
-    Py_DECREF(x);
+#define DATETIME_ADD_MACRO(dict, c, value)              \
+    do {                                                \
```

Comment:
You must add `PyObject *value = (value_expr);` and rename value parameter to value_expr.

Otherwise, DATETIME_ADD_MACRO(d, "resolution", new_delta(0, 0, 1, 0)) calls new_delta() 3 times, since the preprocessor simply replace "value" with "new_delta(0, 0, 1, 0)". It's a dummy text processor.